src/AsyncTimer.cpp

```diff
@@ -73,6 +73,14 @@ void AsyncTimer::cancelAll(bool includeIntervals) {
   }
 }
 
+unsigned long AsyncTimer::getRemaining(unsigned short id) {
+  unsigned long now = millis();
+  for (unsigned short i = 0; i < m_maxArrayLength; i++)
+    if (m_callsArray[i].id == id)
+      return m_callsArray[i].timestamp + m_callsArray[i].delayByMs - now;
+  return -1;

```

Comment:
return 0 instead of -1 because `unsigned long` doesn't support negatives

Reply:
Also beware that `now` might happen to be **after** `m_callsArray[i].timestamp + m_callsArray[i].delayByMs` just because the code so far has been executing synchronously long enough to not allow the library to execute the scheduled callback on time (no callback is guaranteed to be executed at its scheduled time exactly).
Simple example would be if the callback has one more second before it has to execute but the current code takes three seconds before it completes, and at the end of those three seconds `.getRemaining()` will return `-2`

Reply:
Good point, a fix to this problem is to keep the `now` variable and check if `now` is bigger than the sum of the timestamp and the delay, if it is bigger, return 0, otherwise subtract now from it and return the result

Reply:
Thanks for the pointers. I'll update the code